draft-ietf-taps-impl.md

```diff
@@ -963,3 +963,46 @@ If the application indicates a bound on the expected Send or Receive bitrate, an
 
 * Cost Preferences:
 If the application indicates a preference to avoid expensive paths, and some paths are associated with a monetary cost, an implementation should decrease the ranking of such paths. If the application indicates that it prohibits using expensive paths, paths that are associated with a cost should be purged from the decision tree.
+
+# Reasons for errors {#appendix-reasons-errors}
+
+The Transport Services API {{I-D.ietf-taps-interface}} allows for the seven generic error types to specify a more detailed reason as to why an error occurred. This appendix lists some of the possible reasons.
+
+* BadConfiguration:
+The transport properties and endpoints provided by the application are either contradictory or incomplete. Examples include the lack of a remote endpoint on an active open or using a multicast group address while not requesting a unidirectional receive.
+
+* NoCandidates:
+None of the available transport protocols can satisfy the transport properties provided by the application.
+
+* Unresolvable:
+The remote or local specifier provided by the application can not be resolved.
+
+* CantEstablish:
+The TAPS system was unable to establish a transport-layer connection to the remote endpoint specified by the application.
+
+* PolicyProhibited:
+The system policy prevents the TAPS system from performing the action requested by the application. 
+
+* NotCloneable:
+The protocol stack is not capable of being cloned. 
+
+* MessageToLarge:
+The message size is too big for the taps system to handle.
+
+* ProtocolFailure:
+The underlying protocol stack failed. 
+
+* BadMessageProperties:
+The message properties are either contradictory to the transport properties or they can not be satisfied by the TAPS system. 
+
+* BadReceive:
```

Comment:
What does this case mean?

Reply:
While the underlying transport was able to receive everything properly, the transport system implementation itself was not. I wanted to keep the two distinct but that might not be necessary.

Reply:
Does it make sense to split this case between "LocalReceivFailure" und "DeframingFailure"?

Reply:
Another suggestion: ProtocolReceiveFail and SystemReceiveFail.
Or get rid of the distinction here (it's just a suggestion anyway) and just go with ReceiveFailed, which would be in analogy to ResolutionFailed and EstablishmentFailed.

Reply:
I guess we should either have a distinction or not include the example. If we have ReceiveError and the reason is ReceiveFailed I am not sure it provides much extra information? Perhaps ProtocolReceiveFailure and DeframingFailure if we want to include something for ReceiveFailed.

Reply:
And perhaps Failed rather than Failure to be consistent across error reasons.